tools/dotnet-linker/Steps/GenerateReferencesStep.cs

```diff
@@ -0,0 +1,74 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+using Mono.Cecil;
+
+using Xamarin.Bundler;
+using Xamarin.Linker;
+
+namespace Xamarin {
+
+	public class GenerateReferencesStep : ConfigurationAwareStep {
+		private Symbols required_symbols = new Symbols ();
+		
+		protected override string Name { get; } = "Generate References";
+		protected override int ErrorCode { get; } = 2320;
+
+		protected override void TryEndProcess ()
+		{
+			base.TryEndProcess ();
+
+			var items = new List<MSBuildItem> ();
+			var file = Path.Combine (Configuration.CacheDirectory, $"references.mm");
+			if (Configuration.Target.GenerateReferencingSource (file, required_symbols) != null) {
+				var item = new MSBuildItem { Include = file };
+				items.Add (item);
+			}
+			Configuration.WriteOutputForMSBuild ("_ReferencesFile", items);
+		}
+
+		protected override void TryProcessAssembly (AssemblyDefinition assembly)
+		{
+			base.TryProcessAssembly (assembly);
+
+			if (!assembly.MainModule.HasTypes)
+				return;
+
+			var hasSymbols = false;
+			if (assembly.MainModule.HasModuleReferences) {
+				hasSymbols = true;
+			} else if (assembly.MainModule.HasTypeReference ("Foundation.FieldAttribute")) {
+				hasSymbols = true;
+			}
```

Comment:
could we just do this with an or and do it in the above line? We remove the code branches and we just do later a return according to the value.
```csharp
hasSymbols = assembly.MainModule.HasModuleReferences || assembly.MainModule.HasTypeReference ("Foundation.FieldAttribute");
```

Reply:
Sure, I copy & pasted this part from the old mmp code.